datasetinsights/estimators/base.py

```diff
@@ -35,17 +33,18 @@ def create_estimator(
     # todo this makes it so that we lose the tensorboard
     #  writer of non-master processes which could make debugging harder
 
-    writer = SummaryWriter(tb_log_dir)
     kfp_writer = KubeflowPipelineWriter(
-        filename=kfp_metrics_filename, filepath=kfp_metrics_dir,
+        tb_log_dir=tb_log_dir,
+        filename=kfp_metrics_filename,
+        filepath=kfp_metrics_dir,
     )
+    kfp_writer.create_tb_visualization_json()
```

Comment:
Why do we need to call this when the `KubeflowPipelineWriter` is initialized? Can you hide this so that users does not have to remember calling this? How is this different from `KubeflowPipelineWriter.write_metric()`? Should they be coupled into the same writer?

Reply:
`write_metrci()` creates JSON file for metrcis while this creates JSON for tensorboard visualizer. I think they should be coupled to the same writer because `KubeflowPipelineWriter ` shuold handle anything that is related to KFP pipeline/ dashboard.

Reply:
`kfp_writer.create_tb_visualization_json()` we can hide this and instead call when the writer is initialized so that the user don't have to worry about it.